### DISCUSSION TITLE
Combining two sets of meshs with overlapping.

### DISCUSSION THREAD

**Discussion by Meghasemi-96** [idaholab/moose#27269] [Q&A Meshing]
Hello, 

I am having trouble combining two sets of mesh models that I extracted from images. Both sets are the same size and represent different parts of an object, but they share a common region in both images. As shown in attached file, my original structure has three revgions. To have a correct mesh, I divided the structure to two parts (righ hand side image of attached files). I got two separate meshs (each 1000 elements) for these images and then I combineed them, followed by renaming the block IDs. After the combing it seems that two sets of meshes are not merged together and the number of elements were doubled (2000 elements). How can I correct it?

Thanks. 

>Following is the code:

```
[Mesh]
    [generate]
    type = FileMeshGenerator
    file = image_2d_out1.e
    []
    [image]
      type = ImageSubdomainGenerator
      input = generate
      threshold = 90
      file = 1.png
    []
    [interface]
      type = SideSetsBetweenSubdomainsGenerator
      input = image
      primary_block = 0
      paired_block = 1
      new_boundary = 10
    []
    [generate1]
      type = FileMeshGenerator
      file = image_2d_out2.e
      []
      [image1]
        type = ImageSubdomainGenerator
        input = generate1
       threshold = 90   
      file = 2.png
      []
      [interface1]
        type = SideSetsBetweenSubdomainsGenerator
        input = image1
        primary_block = 0
        paired_block = 1
        new_boundary = 10
      []
    [next1]
      type = RenameBlockGenerator
      input = interface1
      old_block = '0 1'
      new_block = '0 2'
    []
    [cmn]
      type = CombinerGenerator
      inputs = 'interface next1'
    []
  []
  [Variables]
    [u]
    []
  []
  [Problem]
    type = FEProblem
    solve = false
  []
  [Executioner]
    type = Steady
  []
  [Outputs]
    file_base = image_meshsecond
    execute_on = 'timestep_end'
    exodus = true
  []
```

  image_2d_out1.e and  image_2d_out2.e were also generated by this code (I checked and they were correct):

```
[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 100
    ny = 100  []
  [image]
    input = gen
    type = ImageSubdomainGenerator
    file = 2.png
    file_suffix = png
    threshold = 90
  []
[]
[Variables]
  [u]
  []
[]
[Problem]
  type = FEProblem
  solve = false
[]
[Executioner]
  type = Transient
  num_steps = 1
  dt = 0.1
[]
[Outputs]
  exodus = true
[]
```

![Picture](https://github.com/idaholab/moose/assets/143123039/0cf17ec1-1098-4c03-8233-e7efdc92ef0b)

**Comment by GiudGiud** [comment]
Hello

The overlap must be removed from the mesh, you can’t have overlapping elements during the simulation.
The meshRepairGenerator might be able to help, once the mesh is stitched properly. 
https://mooseframework.inl.gov/source/meshgenerators/MeshRepairGenerator.html

you ll want to use a StitchedMeshGenerator instead of the combiner to make sure the meshes get properly attached

Guillaume
